### DISCUSSION TITLE
Transform2D set_origin binding

### DISCUSSION THREAD

**Discussion by theia-ajax** [godotengine/godot-proposals#5696] [2D]
There's no gdscript binding to set_origin on Transform2D... I'd like there to be one.

**Comment by aaronfranke** [comment]
What prevents you from using `.origin =`?

In general we don't want to have many methods that mutate the struct, [it can lead to unexpected bugs](https://github.com/godotengine/godot/issues/30744). The `set_rotation`, `set_scale`, and `set_skew` methods are exceptions to this, but note that they are not present in 3D. I don't want to propose we remove those methods, but I would be skeptical of adding more methods like them.

**Reply by theia-ajax** [reply]
I think I tried `.origin` at one point and had an issue but realized I had probably called it on a node itself and not the transform of the node. I found a work-around for my specific issue at the time but just took another pass trying to use transform.origin and it worked as expected so no binding necessary.

Ultimately the issue was ignorance :D
